pkg/activator/handler/handler.go

```diff
@@ -200,7 +199,19 @@ func (a *ActivationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// rewriteHost removes the `Host` header from the inbound request
+// and replaces it with our custom header. This is done to avoid
+// Istio Host based routing.
+// Queue-Proxy will execute the reverse process.
+func rewriteHost(r *http.Request) {
```

Comment:
Maybe we want a helper in `network` to do/undo this?

Reply:
Not a fan, since there's no re-use, but I can see the benefit of having them next to each other for maintability, so moved it. :crab: